### ISSUE TITLE
Submitting an add-on went terrible

### ISSUE THREAD

**Issue by andymckay** [mozilla/addons-server#4700]
Tried to upload a webextension: https://addons.mozilla.org/en-US/developers/addon/close-duplicate-tabs4

It failed saying it "couldn't parse manifest.json".

So I tried again. Then it passed that step, but then failed saying I'd already got that addon slug loaded.

I went to my list of add-ons and it said I had two add-ons partially completed.

I deleted one, went back to the first. Uploaded it, entered a custom slug because it said my slug was taken.

It then uploaded and put my add-on into "Disabled by Mozilla" state.

It completely ignored the custom slug I entered.

Not sure what's going to happen now.

**Comment by eviljeff** [comment]
The add-on looks like it's missing all the required metadata (summary, categories, version license).  Did you add those?

**Comment by eviljeff** [comment]
possible dupe to #4607

**Comment by muffinresearch** [comment]
FWIW: The could not parse the manifest is coming from here: according to kibana: https://github.com/mozilla/addons-server/blob/master/src/olympia/devhub/utils.py#L258 

`[u'Could not parse the manifest file.'] :/data/olympia/src/olympia/devhub/utils.py:259`

**Comment by muffinresearch** [comment]
Digging back further through the logs looks like the parsing of the manifest file error came from the zip lib blowing up: `raise BadZipfile, "File is not a zip file"#012BadZipfile: File is not a zip file` so the error "Could not parse the manifest file" is a bit of a misnomer.

Kibana Timestamp is 2017-02-16T22:51:58 to find logs around that time.

**Comment by andymckay** [comment]
I filled out the metadata a couple of times as I fought past the repeated slug problems. I'll do it again.

**Comment by eviljeff** [comment]
The slug problems are a known bug (I didn't think it was _that_ bad though, so didn't prioritize it).  I think we have two separate issues here - 1) the 'broken' xpi (file system issue?) 2) shaky slug javascript

**Comment by diox** [comment]
regardless of those issues though, how did the metadata disappear ? And what code changed the status of the only file to disabled ?

**Comment by andymckay** [comment]
I cannot set categories:

<img width="764" alt="screenshot 2017-02-17 08 56 37" src="https://cloud.githubusercontent.com/assets/74699/23075723/6f0ed578-f4f2-11e6-96aa-f05297fc084f.png">

**Comment by muffinresearch** [comment]
@eviljeff @diox are there concrete bugs that we can break out of this? It would be good to close this if we can file or point to specific bugs for each part that played it's part in creating this experience.

**Comment by diox** [comment]
Best I can tell there are 3 separate issues being reported here:
- Initial ["Could not parse the manifest file."](https://github.com/mozilla/addons-server/blob/b6db781146f9d18b20baff73800dfab0486bd722/src/olympia/files/utils.py#L656) ; This is an error when reading the zip file. AFAIK we haven't had any other complaints about this, it could be a temporary issue - we were deploying at the time this problem occurred - From looking at kibana we're seeing this from time to time but since it can happen when the manifest file is missing, it's hard to conclude anything from that.
- The slug issue. That's #4607 I believe. @eviljeff might know more, I keep forgetting about what the exact problem is.
- The categories issue. This needs more investigation / some STRs and a new issue.

**Comment by eviljeff** [comment]
The slug issue is (iirc) an addon named `name` getting a slug of `name1`,
because `name` exists; but then the page JS seeing the name is `name` and
changing the slug to `name`, which then causes an error on submission
(obvs). It's possible this only happens on the second attempt (e.g. you
have some field error on the first submit of the page)

The categories bug I'm not sure about, but (random idea) maybe having no
latest version (after you disable the version by aborting submission) borks
the category selection because it doesn't know what Application your add-on
supports so can't offer relevant categories.

**Comment by diox** [comment]
I updated #4607 and filed https://github.com/mozilla/addons-server/issues/5082. The initial issue with the "could not parse the manifest file" remains but I believe it was just a temporary issue ; fixing the other 2 would unblock developers ending in this situation, so I'm closing this bug.
